File: November2021/C++/Single_Element_in_a_SortedArray.cpp
//author : shyam2520
// https://leetcode.com/problems/single-element-in-a-sorted-array/
class Solution {
public:
    int singleNonDuplicate(vector<int>& nums) {
        int l=0,r=nums.size()-1;
        int mid;
        while(l<r)
        {
            mid=l+(r-l)/2;
            if(mid%2)
            {
                if(nums[mid-1]==nums[mid])
                    l=mid+1;
                else
                    r=mid;
            }
            else 
            {
                if(nums[mid+1]==nums[mid])
                    l=mid+1;
                else
                    r=mid;
            }
            
        }
        return nums[l];
    }
};

File: November2021/SingleElementinaSortedArray.java
// TC : O(logn)
// SC : O(1)
class Solution {
	public int singleNonDuplicate(int[] nums) {
		int low = 0, high = nums.length-1;

		while (low < high) {
			int  mid = (low + high) / 2;
			if ((mid % 2 ==1 && nums[mid - 1] == nums[mid]) || ((mid % 2)==0 && nums[mid] == nums[mid + 1])) low = mid + 1;
			else high = mid;
		}
		return nums[low];
	}
}